hazelcast/src/test/java/com/hazelcast/internal/nearcache/AbstractNearCacheBasicTest.java

```diff
@@ -220,6 +221,60 @@ private long getExpectedHitsWithLocalUpdatePolicy(NearCacheTestContext<Integer,
     }
 
     /**
+     * Checks that the Near Cache is populated when {@link DataStructureAdapter#putIfAbsent(Object, Object)} is used.
+     */
+    @Test
+    public void whenPutIfAbsentIsUsed_thenNearCacheShouldBePopulated_withUpdateOnNearCacheAdapter() {
+        whenPutIfAbsentIsUsed_thenNearCacheShouldBePopulated(true, false);
+    }
+
+    /**
+     * Checks that the Near Cache is populated when {@link DataStructureAdapter#putIfAbsent(Object, Object)} is used.
+     */
+    @Test
+    public void whenPutIfAbsentIsUsed_thenNearCacheShouldBePopulated_withUpdateOnDataAdapter() {
+        whenPutIfAbsentIsUsed_thenNearCacheShouldBePopulated(false, false);
+    }
+
+    /**
+     * Checks that the Near Cache is populated when {@link DataStructureAdapter#putIfAbsentAsync(Object, Object)} is used.
+     */
+    @Test
+    public void whenPutIfAbsentAsyncIsUsed_thenNearCacheShouldBePopulated_withUpdateOnNearCacheAdapter() {
+        whenPutIfAbsentIsUsed_thenNearCacheShouldBePopulated(true, true);
+    }
+
+    /**
+     * Checks that the Near Cache is populated when {@link DataStructureAdapter#putIfAbsentAsync(Object, Object)} is used.
+     */
+    @Test
+    public void whenPutIfAbsentAsyncIsUsed_thenNearCacheShouldBePopulated_withUpdateOnDataAdapter() {
+        whenPutIfAbsentIsUsed_thenNearCacheShouldBePopulated(false, true);
+    }
+
+    private void whenPutIfAbsentIsUsed_thenNearCacheShouldBePopulated(boolean useNearCacheAdapter, boolean useAsync) {
+        NearCacheTestContext<Integer, String, NK, NV> context = createContext();
+        DataStructureAdapter<Integer, String> adapter = useNearCacheAdapter ? context.nearCacheAdapter : context.dataAdapter;
+        if (useAsync) {
+            assumeThatMethodIsAvailable(adapter, "putIfAbsentAsync", Object.class, Object.class);
```

Comment:
I'm already using `assertNearCacheSizeEventually()`, but the `owned entry count` doesn't seem to be updated to the correct value at all.